packages/memory-inspector/src/browser/utils/memory-widget-manager.ts

```diff
@@ -0,0 +1,159 @@
+    @inject(ApplicationShell) protected readonly shell: ApplicationShell;
+    @inject(MessageService) protected readonly messageService: MessageService;
+
+    protected readonly onNewWidgetCreated = new Emitter<MemoryWidget>();
+    readonly onDidCreateNewWidget = this.onNewWidgetCreated.event;
+
+    protected readonly onSelectedWidgetChanged = new Emitter<MemoryWidget | undefined>();
+    readonly onDidChangeSelectedWidget = this.onSelectedWidgetChanged.event;
+
+    protected readonly onChangedEmitter = new Emitter<void>();
+    readonly onChanged = this.onChangedEmitter.event;
+
+    protected readonly _availableWidgets = new Map<string, MemoryWidget>();
+    protected _focusedWidget: MemoryWidget | undefined;
+    protected _canCompare = false;
+
+    get availableWidgets(): MemoryWidget[] {
+        return Array.from(this._availableWidgets.values());
+    }
+
+    get canCompare(): boolean {
+        return this._canCompare;
+    }
+
+    @postConstruct()
+    protected init(): void {
+        this.toDispose.pushAll([
+            this.shell.onDidChangeActiveWidget(({ newValue }) => {
+                if (newValue instanceof MemoryWidget) {
+                    this._focusedWidget = newValue;
+                }
+            }),
+            this.widgetManager.onDidCreateWidget(e => {
+                const { widget } = e;
+                if (widget instanceof MemoryWidget) {
+                    this._availableWidgets.set(widget.id, widget);
+                    this.toDispose.push(widget.onDidDispose(() => {
+                        this._availableWidgets.delete(widget.id);
+                        if (widget === this._focusedWidget) {
+                            this.focusedWidget = undefined;
+                        }
+                        this.onChangedEmitter.fire();
+                    }));
+                }
+            }),
+            this.onChanged(() => this.setCanCompare()),
+            this.onNewWidgetCreated,
+            this.onChangedEmitter,
+            this.onSelectedWidgetChanged,
+        ]);
+    }
+
+    get focusedWidget(): MemoryWidget | undefined {
+        return this._focusedWidget ?? this._availableWidgets.values().next().value;
+    }
+
+    set focusedWidget(title: MemoryWidget | undefined) {
+        this._focusedWidget = title;
+        this.onSelectedWidgetChanged.fire(title);
+    }
+
+    protected setCanCompare(): void {
+        this._canCompare = this.availableWidgets.filter(widget => !RegisterWidget.is(widget) && !MemoryDiffWidget.is(widget)).length > 1;
+    }
+
+    async createNewMemoryWidget<T extends MemoryWidget>(kind: 'register' | 'memory' = 'memory'): Promise<T> {
+        this.widgetDisplayId = this._availableWidgets.size !== 0 ? this.widgetDisplayId + 1 : 1;
+        const options: MemoryWidgetOptions = { identifier: this.createdWidgetCount += 1, displayId: this.widgetDisplayId };
+        const widgetId = kind === 'memory'
```

Comment:
This is reasonable, but doing it right would require a bit of a refactor - I'll leave this one out for now.

Reply:
Sure, you can keep the effort as low as possible. I just thought something simple along the lines of:
```
protected mapKindToWidgetId(kind: 'register' | 'memory' = 'memory') {
   return kind === 'memory' ? MemoryWidget.ID : RegisterWidget.ID;
}
```

So in case you want the editable memory widget you could simply do:

```
protected override mapKindToWidgetId(kind: 'register' | 'memory' = 'memory') {
   return kind === 'memory' ? EditableMemoryWidget.ID : RegisterWidget.ID;
}
```